src/main/java/it/smartcommunitylab/aac/spid/registry/IconStore.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2024 the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package it.smartcommunitylab.aac.spid.registry;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+import net.shibboleth.utilities.java.support.collection.Pair;
+import net.shibboleth.utilities.java.support.net.URLBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.lang.Nullable;
+
+/*
+ * IconStore is a utility class used to recover the icon associated to a
+ * SPID identity provider. Icon URLs are obtained from a request with a request
+ * with pattern
+ *  https://registry.spid.gov.it/entities-idp/{entityId}?output=json
+ */
+public class IconStore {
```

Comment:
this should be refactored:
* sourcing metadata from a remote address is a security risk and should be handled adequately handled
* we should avoid using URLs received from a remote party as img sources, they could link to scripts etc.
* this implementation uses lots of low level or strange classes